llama_index/schema.py

```diff
@@ -112,10 +215,110 @@ def get_embedding(self) -> List[float]:
             raise ValueError("embedding not set.")
         return self.embedding
 
+
+class TextNode(BaseNode):
+    content: str
+    start_char_idx: Optional[int] = Field(
+        default=None, description="Start char index of the node."
+    )
+    end_char_idx: Optional[int] = Field(
+        default=None, description="End char index of the node."
+    )
+    text_template: str = Field(
+        default=DEFAULT_TEXT_NODE_TMPL,
+        description=(
+            "Template for how text is formatted, with {content} and "
+            "{metadata_str} placeholders."
+        ),
+    )
+    metadata_template: str = Field(
+        default=DEFAULT_METADATA_TMPL,
+        description=(
+            "Template for how metadata is formatted, with {key} and "
+            "{value} placeholders."
+        ),
+    )
+    metadata_seperator: str = Field(
+        default="\n",
+        description="Seperator between metadata fields when converting to string.",
+    )
+
+    def _generate_node_hash(self) -> str:
+        """Generate a hash to represent the node."""
+        doc_identity = str(self.content) + str(self.metadata_str)
+        return sha256(doc_identity.encode("utf-8", "surrogatepass")).hexdigest()
+
+    @classmethod
+    def get_type(cls) -> str:
+        """Get Object type."""
+        return ObjectType.TEXT
+
     @property
-    def extra_info_str(self) -> Optional[str]:
-        """Extra info string."""
-        if self.extra_info is None:
-            return None
+    def is_content_none(self) -> bool:
+        """Check if content is None."""
+        return self.content == ""
+
+    def get_content(self) -> str:
+        """Get object content."""
+        metadata_str = self.metadata_str()
+        return self.text_template.format(
+            content=self.content, metadata_str=metadata_str
+        )
+
+    def metadata_str(self) -> str:
+        """Convert metadata to a string."""
+
+        return self.metadata_seperator.join(
+            [
+                self.metadata_template.format(key=key, value=str(value))
+                for key, value in self.metadata.items()
+                if self.usable_metadata is None or key in self.usable_metadata
+            ]
+        )
+
+    def get_node_info(self) -> Dict[str, Any]:
```

Comment:
just fyi, there was another (abused) use-case of node_info to keep track of any additional metadata that's not flat simple types.